migrations/20201122043349_more-mod-data.sql

```diff
@@ -0,0 +1,54 @@
+CREATE TABLE donation_platforms (
+    id serial PRIMARY KEY,
+    short varchar(100) UNIQUE,
+    name varchar(500) UNIQUE
+);
+
+INSERT INTO donation_platforms (short, name) VALUES ('patreon', 'Patreon');
+INSERT INTO donation_platforms (short, name) VALUES ('bmac', 'Buy Me a Coffee');
+INSERT INTO donation_platforms (short, name) VALUES ('paypal', 'PayPal');
+INSERT INTO donation_platforms (short, name) VALUES ('github', 'GitHub Sponsors');
+INSERT INTO donation_platforms (short, name) VALUES ('ko-fi', 'Ko-fi');
+INSERT INTO donation_platforms (short, name) VALUES ('other', 'Other');
+
+CREATE TABLE mods_donations (
+    joining_mod_id bigint REFERENCES mods ON UPDATE CASCADE NOT NULL,
+    joining_platform_id int REFERENCES donation_platforms ON UPDATE CASCADE NOT NULL,
+    url varchar(2048) NOT NULL,
+    PRIMARY KEY (joining_mod_id, joining_platform_id)
+);
+
+CREATE TABLE side_types (
+    id serial PRIMARY KEY,
+    name varchar(64) UNIQUE
+);
+
+INSERT INTO side_types (name) VALUES ('required');
+INSERT INTO side_types (name) VALUES ('no-functionality');
+INSERT INTO side_types (name) VALUES ('unsupported');
+
+CREATE TABLE licenses (
+    id serial PRIMARY KEY,
+    short varchar(60) UNIQUE,
+    name varchar(1000) UNIQUE
+);
+
+INSERT INTO licenses (short, name) VALUES ('arr', 'All Rights Reserved');
+
+ALTER TABLE versions
+    ADD COLUMN featured BOOLEAN NOT NULL default FALSE;
+ALTER TABLE files
+    ADD COLUMN "primary" BOOLEAN NOT NULL default FALSE;
+
+ALTER TABLE mods
+    ADD COLUMN license integer REFERENCES licenses NOT NULL default 1;
```

Comment:
This should not have a default, or licenses should have an unspecified variant.  This should be required to be specified in the future, but it's not safe to assume that all current mods are all rights reserved.

Reply:
It is safe to assume that all mods are ARR, unless specified. It also saves us from legal trouble, because if we specify the wrong license for a mod (on our end), that would be bad for us